kafka/avro.go

```diff
@@ -0,0 +1,68 @@
+package kafka
+
+import (
+	"reflect"
+)
+
+type Schema struct {
+	Type   string  `json:"type"`
+	Name   string  `json:"name"`
+	Fields []Field `json:"fields"`
+}
+
+type Field struct {
+	Name string      `json:"name"`
+	Type interface{} `json:"type"`
+}
+
+func createSchema(name string, fieldsMap map[string]interface{}) Schema {
+	fields := make([]Field, 0, len(fieldsMap))
+	for key, val := range fieldsMap {
+		fields = append(fields, createField(key, val))
+	}
+	return Schema{
+		Type:   "record",
+		Name:   name,
+		Fields: fields,
+	}
+}
+
+func createField(key string, val interface{}) Field {
+	var value interface{}
+	if reflect.TypeOf(val).Kind() == reflect.Slice {
+		value = createComplexField(val)
+	} else {
+		value = val
+	}
+	f := Field{
+		Name: key,
+		Type: value,
+	}
+	return f
+}
+
+func createComplexField(val interface{}) interface{} {
```

Comment:
a docstring would improve the readability of the method

Reply:
Agree, the explication of the complex fields is on the docstring of the GetFields function, but a specific one would help as the reflection complexity is high
I'll add it on the next PR :)

Reply:
I finally added it here